ffi/src/scan.rs

```diff
@@ -92,9 +92,9 @@ fn scan_impl(
 ) -> DeltaResult<Handle<SharedScan>> {
     let mut scan_builder = snapshot.scan_builder();
     if let Some(predicate) = predicate {
-        let mut visitor_state = KernelExpressionVisitorState::new();
-        let exprid = (predicate.visitor)(predicate.predicate, &mut visitor_state);
-        let predicate = unwrap_kernel_expression(&mut visitor_state, exprid);
+        let mut visitor_state = KernelExpressionVisitorState::default();
+        let predid = (predicate.visitor)(predicate.predicate, &mut visitor_state);
+        let predicate = unwrap_kernel_expression(&mut visitor_state, predid);
```

Comment:
nit: i read this as pre-did (though maybe that's just me lol) consider rename
```suggestion
        let pred_id = (predicate.visitor)(predicate.predicate, &mut visitor_state);
        let predicate = unwrap_kernel_expression(&mut visitor_state, pred_id);
```